### ISSUE TITLE
"panic: slice bounds out of range" on spillable groupby

### ISSUE THREAD

**Issue by philrz** [brimsec/zq#953]
I took my first shot at verifying spillable groupby using `zq` commit `de55ba7` and managed to crash it. This is with the [Zeek zq-sample-data](https://github.com/brimsec/zq-sample-data/tree/master/zeek-default).

```
~/work/zq-sample-data/zeek-default$ zq -t 'count() by uid' conn.log.gz 
#0:record[uid:bstring,count:uint64]
0:[C0002i1FTbwDjYiDo5;1;]
0:[C000NXHP3tORDvjT8;1;]
...
0:[C1OaS614cz2X852C9j;1;]
0:[C1OaX72K3wC1A6ctppanic: runtime error: slice bounds out of range [:37] with capacity 15

goroutine 20 [running]:
github.com/brimsec/zq/zcode.(*Iter).Next(0xc00082fa20, 0xc00cf2bfe7, 0xa, 0x19, 0x1, 0x0, 0x0)
	/Users/phil/work/zq/zcode/iter.go:30 +0x242
github.com/brimsec/zq/expr.slice(0xc00cf2bfe6, 0x15, 0x1a, 0x1, 0xc0187664a0, 0x1, 0x100acab, 0xc000020000)
	/Users/phil/work/zq/expr/temp.go:47 +0x94
github.com/brimsec/zq/expr.CompileFieldAccess.func1(0xc01c29bb00, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/phil/work/zq/expr/temp.go:30 +0x144
github.com/brimsec/zq/reducer/compile.(*Row).ConsumePart(0xc00082fd28, 0xc01c29bb00, 0x1, 0xc000208b20)
	/Users/phil/work/zq/reducer/compile/row.go:38 +0xe1
github.com/brimsec/zq/proc.merger.func1(0xc01c29bb00, 0xc0120d2b88, 0x0, 0x0, 0xc01c29bbc0, 0x0, 0x0)
	/Users/phil/work/zq/proc/groupby.go:264 +0xb6
github.com/brimsec/zq/proc.(*MergeReader).Read(0xc0003e00d0, 0xc01c29bbc0, 0x0, 0x0)
	/Users/phil/work/zq/proc/combiner.go:142 +0x234
github.com/brimsec/zq/proc.(*GroupByAggregator).readSpills(0xc0003e0000, 0x1, 0x0, 0x0, 0x0, 0x0)
	/Users/phil/work/zq/proc/groupby.go:598 +0xc1
github.com/brimsec/zq/proc.(*GroupByAggregator).Results(0xc0003e0000, 0x1a13c01, 0xc01050df20, 0x0, 0x0, 0x0)
	/Users/phil/work/zq/proc/groupby.go:571 +0x4a
github.com/brimsec/zq/proc.(*GroupBy).run(0xc0002f0dc0)
	/Users/phil/work/zq/proc/groupby.go:345 +0x390
created by github.com/brimsec/zq/proc.(*GroupBy).Pull.func1
	/Users/phil/work/zq/proc/groupby.go:322 +0x3e
```

**Comment by henridf** [comment]
@philrz this bug is fixed, but I would suggest waiting till #961 (and any possible offshoots) are settled before proceeding with further verification.

**Comment by philrz** [comment]
As #961 is merged and I see no other offshoots, so I've now verified in `zq` commit `55b3fc4`. The aggregation I'd initially attempted on the zq-sample-data now completes without error:

```
~/work/zq-sample-data/zeek-default$ zq -t 'count() by uid' conn.log.gz
#0:record[uid:bstring,count:uint64]
0:[C0002i1FTbwDjYiDo5;1;]
0:[C000NXHP3tORDvjT8;1;]
...
0:[CzzzrG420kZYy90Kte;1;]
0:[Czzzwx35Fooc3c4tK3;1;]
```

Thanks @henridf!
